vxyowsup/tests/test_whatsapp.py

```diff
@@ -26,8 +26,12 @@ def getDummyCoreLayers():
 def TUMessage_to_PTNode(message):
     # message is TransportUserMessage
     # returns ProtocolTreeNode
-    return TextMessageProtocolEntity(message['content'], to=message['to_addr']
-                                     + '@s.whatsapp.net').toProtocolTreeNode()
+    if message['to_addr']:
+        return TextMessageProtocolEntity(message['content'], to=message['to_addr']
+                                         + '@s.whatsapp.net').toProtocolTreeNode()
+    elif message['from_addr']:
```

Comment:
I think that's explicitly not the case with `TextMessageProtocolEntity` for some reason.